### PR TITLE
[FEATURE]: Remove upper cap on Python version

### LINKED ISSUE #5194: [FEATURE] Support Python 3.12.x for argilla client 2.0
Cucu! :) First of all huge congrats on the acquirement, just amazing, really look up to you guys!! I hope you could already savor this achievement, and that you are all doing fine! Un fuerte abrazo!

**Is your feature request related to a problem? Please describe.**
I'm on Ubuntu 24.04 and mainly work with venvs. Since the system Python version is 3.12.3 I could not install the argilla2.0 client.

**Describe the solution you'd like**
Support Python versions 3.12.x by removing the upper version cap. There are arguments to be made not to use upper caps: 
- https://packaging.python.org/en/latest/guides/writing-pyproject-toml/#python-requires
- https://discuss.python.org/t/requires-python-upper-limits/12663

**Describe alternatives you've considered**
update the version cap to `<3.13`

**Additional context**

### PR DESCRIPTION
**Pull request by dcfidalgo**
# Pull Request Template
Removed upper Python version cap.

Closes #5194

**Type of change**

- New feature (non-breaking change which adds functionality)

**How Has This Been Tested**

I realized that with the upper cap, the `build (3.12)` action was actually testing Python version 3.10.12  (Ubuntu's 22.04 system Python version). You can check the line here: 
- https://github.com/argilla-io/argilla/actions/runs/9897045807/job/27340638847#step:7:14

So this PR should test for version 3.12.4

**Checklist**

- I added relevant documentation
- I followed the style guidelines of this project
- I did a self-review of my code
- I made corresponding changes to the documentation
- I confirm My changes generate no new warnings
- I have added tests that prove my fix is effective or that my feature works
- I have added relevant notes to the CHANGELOG.md file (See https://keepachangelog.com/)

### CODE REVIEW

File: argilla/pyproject.toml [hunk 1]
```diff
@@ -4,7 +4,7 @@ description = "The Argilla python server SDK"
 authors = [
     {name = "Argilla", email = "contact@argilla.io"},
 ]
-requires-python = ">=3.9,<=3.12"
+requires-python = ">= 3.9"
```
Comment (frascuchon) [inline argilla/pyproject.toml]: Thanks for this @dcfidalgo!!! (and happy to see you here :-)) I think this explicit upper limit also works and avoids installing for unsupported versions.
```suggestion
requires-python = ">= 3.9, <3.13"
```

**Comment by davidberenstein1957** [comment]
@dcfidalgo thank you for the PR and kind comment w.r.t. the acquisition :) We are getting settled in nicely.

There is a pyarrow installation failure during the cicd for python 3.12, which might be able to be resolved with versioing. https://github.com/argilla-io/argilla/actions/runs/9898629294/job/27345861980

File: argilla/pyproject.toml (hunk 1 above)
Comment (dcfidalgo) [reply argilla/pyproject.toml]: Yes, this also works for me! :) Not sure where the pyarrow installation error comes from, it should support 3.12 ...

**Comment by frascuchon** [comment]
> @dcfidalgo thank you for the PR and kind comment w.r.t. the acquisition :) We are getting settled in nicely.
> 
> There is a pyarrow installation failure during the cicd for python 3.12, which might be able to be resolved with versioing. https://github.com/argilla-io/argilla/actions/runs/9898629294/job/27345861980

https://github.com/apache/arrow/issues/37880

Not sure about which package is bringing this package version. Maybe we should fix the dev/ci environments to 14.x

**Review by frascuchon — APPROVED** [review]
I will merge it. We need to:

1. `pdm install` to refresh lock file
2. `pdm update pyarrow` to refresh pyarrow version

I will create a new PR for that. Thanks @dcfidalgo !!

**Pull request merged**